app/controllers/items/detail.js

```diff
@@ -254,6 +264,54 @@ export default GoodcityController.extend(
         this.toggleProperty("showSetList");
       },
 
+      /**
+       * Removes an item from a box/pallet
+       * @param { Item } pkg The package we wish to remove from the box/pallet
+       */
+      selectLocationAndUnpackItem() {
+        const location = this.get("locationService.selectedLocation");
+        let item = this.get("removableItem");
+        if (!location) {
+          return false;
+        }
+        if (item) {
+          const params = {
+            item_id: item.id,
+            location_id: location.id,
+            task: "unpack",
+            quantity: item.added_quantity
+          };
+          this.get("packageService").addRemoveItem(this.get("item.id"), params);
+        }
+        this.send("fetchAssociatedPackages");
+      },
+
+      /**
+       * Fetches all the assoicated packages to a box/pallet
+       */
+      fetchAssociatedPackages() {
+        this.showLoadingSpinner();
+        return this.get("packageService")
+          .fetchAssociatedPackages(this.get("item.id"))
+          .then(data => {
+            this.store.pushPayload(data);
+            this.set("associatedPackages", data.items);
+          })
+          .finally(() => {
+            this.hideLoadingSpinner();
+          });
+      },
+
+      openLocationSearch(item) {
+        this.set("locationService.openLocationSearch", true);
```

Comment:
The way to use the location service is : 

```javascript
const selectedLocation = await this.get('locationService').userPickLocation( /* opts */ );
// do something with the selected location
```
We shouldn't store anything in the component, or modify the service's properties directly. That's a dangerous game